server/controllers/email-report/weeklySummaryReport.js

```diff
@@ -0,0 +1,213 @@
+  var week = period.periods.week.limit;
+
+  var monday = '2016-01-01 00:00:00.000';// week.start();
+  var sunday = '2016-12-31 23:59:59.999';// week.end();
+
+  var values = [monday, sunday];
+
+
+  return db.exec(sql, values);
+
+}
+
+function loadPatientData() {
+
+  var period = new Period(new Date());
+
+
+  var week = period.periods.week.limit;
+
+  var monday = week.start();
+  var sunday = week.end();
+
+
+  var Lastweek = period.periods.lastWeek.limit;
+
+  var lastmonday = Lastweek.start();
+  var lastsunday = Lastweek.end();
+
+
+  //patients registered this week
+
+  var sql = `
+          SELECT t1.NumberOfpatientsThisWeek,t2.NumberOfpatientsLastWeek
+           FROM 
+          (
+            SELECT count(uuid) as NumberOfpatientsThisWeek
+            FROM patient
+            WHERE (registration_date BETWEEN ? AND ? )
+          ) as t1,
+          (
+            SELECT count(uuid) as NumberOfpatientsLastWeek
+            FROM patient
+            WHERE (registration_date BETWEEN ? AND ? )
+          ) as t2
+          `;
+
+  var values = [monday, sunday, lastmonday, lastsunday];
+
+  var thisWeekPatients = [];
+
+  return db.exec(sql, values);
+
+}
+
+
+//first and last invoice per day for a week
+function MaxMinInvoiceDate() {
+
+  var sql = `SELECT  MIN(i.date) as minDate, MAX(i.date) as maxDate
+              From invoice i
+              WHERE DATE(i.date) BETWEEN ? AND ?
+              GROUP BY DATE(i.date)`;
+
+  var period = new Period(new Date());
+  var week = period.periods.week.limit;
+
+  var monday = week.start();
+  var sunday = week.end();
+
+  return db.exec(sql, [monday, sunday]);
+}
+
+function view1(req, res, next) {
+
+  const options = {
+    renderer: 'pdf',
+    saveReport: false,
+    filename: 'test_summary_report',
+    orientation: 'portrait',
+  };
+
+  //patient registered
+  loadPatientData().then(_patientsData => {
+
+    //services incomes
+    loadServicesIncome().then(_servicesIncome => {
+
+      //max and min date of invoice each day for a week
+      MaxMinInvoiceDate().then(_MaxMinInvoiceDate => {
+
+        //rendering the report
+        var report = new ReportManager(TEMPLATE, req.session, options);
+
+        if (_patientsData.lenght === 0) {
+          _patientsData = { "NumberOfpatientsThisWeek": 0, "NumberOfpatientsLastWeek": 0 };
+        } else {
+          _patientsData = _patientsData[0];
+        }
+
+        var data = {
+          patientsData: _patientsData,
+          servicesIncome: _servicesIncome,
+          MaxMinInvoiceDate: _MaxMinInvoiceDate
+        }
+
+        report.render(data)
+          .then((result) => {
+            res.set(result.headers).send(result.report);
+          });
+      });
+    });
+
+  })
+    .catch(next)
+    .done();
+
+
+}
+
+
+//one of the reports that bhima send by email
+function weeklySummaryReport(req, res, next) {
+
+
+  const options = {
+    renderer: 'pdf',
+    saveReport: false,
+    filename: 'test_summary_report.pdf',
```

Comment:
The file name here could be formatted as the current weeks time stamp - it looks like the filename here was just for testing purposes.